server/src/test/java/org/elasticsearch/common/io/stream/BytesStreamsTests.java

```diff
@@ -970,4 +970,30 @@ boolean failOnTooManyNestedExceptions(Throwable throwable) {
         assertThat(newEx.getMessage(), equalTo("disk broken"));
         assertArrayEquals(newEx.getStackTrace(), rootEx.getStackTrace());
     }
+
+    public void testWriteCircularReferenceSuppressedException() throws IOException {
```

Comment:
From what I tested, this test will pass without the fix

Reply:
Doh, this fails with `ElasticsearchException`

Reply:
Updated